stdlib/public/stubs/LibcShims.cpp

```diff
@@ -213,3 +225,77 @@ swift::_swift_stdlib_cxx11_mt19937_uniform(__swift_uint32_t upper_bound) {
   std::uniform_int_distribution<__swift_uint32_t> RandomUniform(0, upper_bound);
   return RandomUniform(getGlobalMT19937());
 }
+
+#if defined(__APPLE__)
+#include "TargetConditionals.h"
+#if defined(TARGET_IPHONE_SIMULATOR) \
+  || ( defined(TARGET_OS_IPHONE) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 10000 ) \
+  || ( defined(TARGET_OS_MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 101200 )
+SWIFT_RUNTIME_STDLIB_INTERFACE
+void swift::_swift_stdlib_random(void *buf,
+                                __swift_ssize_t nbytes,
+                                __swift_uint32_t debug_flags) {
+  arc4random_buf(buf, nbytes);
+}
+#else
+#if defined(TARGET_OS_IPHONE)
+#include <Security/Security.h>
+SWIFT_RUNTIME_STDLIB_INTERFACE
+void swift::_swift_stdlib_random(void *buf,
+                                __swift_ssize_t nbytes,
+                                __swift_uint32_t debug_flags) {
+  if (SecRandomCopyBytes(kSecRandomDefault, nbytes, buf) != 0) {
```

Comment:
`SecRandomCopyBytes()` should be available on all of the Apple platforms supported by Swift. I don't think you need the `arc4random()` or `/dev/urandom` implementations there.

Reply:
arc4random was suggested by Ben Cohen on the evolution mailing list here: https://lists.swift.org/pipermail/swift-evolution/Week-of-Mon-20171002/040195.html . I can substitute `SecRandomCopyBytes()` instead of reading `/dev/urandom`.

Reply:
My understanding is that `SecRandomCopyBytes()` is cryptographically superior to `arc4random()` on old Apple OS versions. I see no reason not to use it on all versions.

Reply:
`arc4random(3)` was replaced in macOS 10.12 and iOS 10 to provide a more secure generator. I do indeed have the version checks to make sure to use `arc4random(3)` on those OS versions on and `SecRandomCopyBytes()` for older versions.

Reply:
* Should `arc4random_buf` vs `SecRandomCopyBytes` use [__builtin_available](https://clang.llvm.org/docs/LanguageExtensions.html#objective-c-available)?

* Should `__swift_ssize_t` be `__swift_size_t` instead?

* Can the `debug_flags` parameter be removed?
  Other APIs always call `fatalError` with zero flags.

```cpp
#if defined(__APPLE__)
#include <Security/Security.h>
SWIFT_RUNTIME_STDLIB_INTERFACE
void swift::_stdlib_random(void *buf, __swift_size_t nbytes) {
  if (__builtin_available(macOS 10.12, iOS 10, tvOS 10, watchOS 3, *)) {
    arc4random_buf(buf, nbytes);
  } else {
    OSStatus status {SecRandomCopyBytes(kSecRandomDefault, nbytes, buf)};
    if (status != errSecSuccess) {
      fatalError(0, "SecRandomCopyBytes failed with error %d\n", status);
    }
  }
}
#endif
```